cirq/optimizers/eject_z_test.py

```diff
@@ -182,7 +184,8 @@ class UnknownGate(cirq.Gate):
             cirq.Moment([cirq.Z(q)]),
             cirq.Moment([u(q)]),
             cirq.Moment(),
-            cirq.Moment([cirq.Y(q)]),
+            cirq.Moment([cirq.Y(q)**-1.0]),
+            cirq.Moment(),
```

Comment:
The output of this test changed?

Reply:
With this current PR; yes.

Reply:
Do you understand why it changed? It's surprising to me.

Reply:
I haven't. I've been trying to figure out what it's testing because of the -1 phase that is added to `cirq.Y(q)`. I haven't gotten to looking where the empty moments are coming from.

Reply:
The test still looks acceptable to me so I'll LGTM with this present.